### ISSUE TITLE
Subresource endpoint

### ISSUE THREAD

**Issue by jpkleemans** [thephpleague/fractal#301]
Lets say I have the following endpoint: `/authors/{author_id}/books/{book_id}`.
How can I create the output for this endpoint with Fractal by keeping the correct links?

I was thinking about something like:
```php
$item = new Item($data, $bookTransformer, 'authors.books');
```

But this outputs `/authors.books/{book_id}` as self-link.

**Comment by lukasoppermann** [comment]
I don't think this is the idea. You are returning a link to the actual resource so just books.

This way you get the correct link of /books/{book_ID}

**Comment by seant23** [comment]
I think this fine to do. I've seen many APIs that do this. It's situational of course but you should be able to do this.

I ended up created a propietaryJsonApiSerializer, that does two things:

1. It allows me to set the resource key in the transformer, simply because having to add in the resource key for every single item/collection call seems really shitty.

2. I added a check for the data array. If it includes _resourceAddress, then I use that instead of the auto generated $resourceKey/$id links.

https://gist.github.com/seant23/2e049f73b3cf9b52635a3cd0b9fc9c50
